packages/atomic/src/components/search/atomic-refine-modal/atomic-refine-modal.tsx

```diff
@@ -126,9 +132,115 @@ export class AtomicRefineModal implements InitializableComponent {
   }
 
+  private createFacetSlot(): HTMLDivElement {
+    const divSlot = document.createElement('div');
+    divSlot.setAttribute('slot', 'facets');
+    this.addElementStyle(divSlot);
+
+    const facets = this.queryCurrentFacets();
+
+    const sortedFacets = sortFacetsViaManager(facets, this.facetManager);
+
+    const {visibleFacets, invisibleFacets} = sortFacetVisibility(
+      sortedFacets,
+      this.bindings.store.getAllFacets()
+    );
+
+    const visibleFacetsClone = this.cloneFacets(visibleFacets);
+    const invisibleFacetsClone = this.cloneFacets(invisibleFacets);
+
+    updateCollapsedState(visibleFacetsClone, this.collapseFacetsAfter);
+
+    divSlot.append(...visibleFacetsClone);
+    divSlot.append(...invisibleFacetsClone);
+
+    const generator = this.makeAutomaticFacetGenerator();
+    if (generator) {
+      updateCollapseFacetsAfter(
+        generator,
+        visibleFacets.length,
+        this.collapseFacetsAfter
+      );
+      divSlot.append(generator);
+    }
+
+    return divSlot;
+  }
+
+  private cloneFacets(facets: BaseFacetElement[]): BaseFacetElement[] {
+    return facets.map((facet) => {
+      facet.classList.remove(popoverClass);
+      facet.setAttribute(isRefineModalFacet, '');
+      const clone = facet.cloneNode(true) as BaseFacetElement;
+      clone.isCollapsed = true;
+      return clone;
+    });
+  }
+
+  private makeAutomaticFacetGenerator() {
+    if (!this.bindings.engine.state.automaticFacetSet?.desiredCount) {
+      return;
+    }
+    const generator = document.createElement(
+      'atomic-automatic-facet-generator'
+    );
+
+    generator.setAttribute(
+      'desired-count',
+      `${this.bindings.engine.state.automaticFacetSet?.desiredCount}`
+    );
+    this.addElementStyle(generator);
+
+    return generator;
+  }
+
+  private queryCurrentFacets(): BaseFacetElement[] {
+    const facetIds = Object.keys(this.bindings.store.getAllFacets());
+    const facetElements: BaseFacetElement[] = [];
+
+    facetIds.forEach((facetIdValue) => {
+      const matchingElement = document.querySelector(
+        `[facet-id="${facetIdValue}"]`
+      );
+      if (matchingElement) {
+        const element = matchingElement as BaseFacetElement;
+        facetElements.push(element);
+      }
+    });
+    return facetElements;
+  }
+
+  public updateCollapse() {
+    const facetContainer = this.host.querySelector('div[slot="facets"]');
+    if (!facetContainer) {
+      return;
+    }
+
+    const facets = getFacetsInChildren(facetContainer.children);
+    const {visibleFacets} = sortFacetVisibility(
+      facets,
+      this.bindings.store.getAllFacets()
+    );
+    const generator = getGeneratorInChildren(facetContainer.children);
+
+    updateCollapsedState(visibleFacets, this.collapseFacetsAfter);
+    updateCollapseFacetsAfter(
+      generator,
+      visibleFacets.length,
+      this.collapseFacetsAfter
+    );
+  }
+
+  private addElementStyle(el: HTMLElement) {
```

Comment:
I think this should have a more descriptive name, like:
```suggestion
  private addFacetColumnStyling(style: CSSStyleDeclaration) {
```